CRM/Contribute/Import/Parser/Contribution.php

```diff
@@ -602,4 +530,105 @@ public function &getImportedContributions() {
+    foreach ($params as $key => $val) {
+      if ($val) {
+        switch ($key) {
+          case 'receive_date':
+            if ($dateValue = CRM_Utils_Date::formatDate($params[$key], $dateType)) {
+              $params[$key] = $dateValue;
+            }
+            else {
+              $errorMessage[] = ts('Receive Date');
+            }
+            break;
+
+          case 'cancel_date':
+            if ($dateValue = CRM_Utils_Date::formatDate($params[$key], $dateType)) {
+              $params[$key] = $dateValue;
+            }
+            else {
+              $errorMessage[] = ts('Cancel Date');
+            }
+            break;
+
+          case 'receipt_date':
+            if ($dateValue = CRM_Utils_Date::formatDate($params[$key], $dateType)) {
+              $params[$key] = $dateValue;
+            }
+            else {
+              $errorMessage[] = ts('Receipt date');
+            }
+            break;
+
+          case 'thankyou_date':
+            if ($dateValue = CRM_Utils_Date::formatDate($params[$key], $dateType)) {
+              $params[$key] = $dateValue;
+            }
+            else {
+              $errorMessage[] = ts('Thankyou Date');
+            }
+            break;
+        }
+      }
+    }
+    return $errorMessage;
+  }
+
+  /**
+   * Format input params to suit api handling.
+   *
+   * Over time all the parts of  _civicrm_api3_deprecated_formatted_param
+   * and all the parts of the import function on this class that relate to
+   * reformatting input should be moved here and tests should be added in
+   * CRM_Contribute_Import_Parser_ContributionTest.
+   *
+   * @param array $params
+   */
+  public function formatInput(&$params) {
+    $dateType = CRM_Core_Session::singleton()->get('dateTypes');
+    $customDataType = !empty($params['contact_type']) ? $params['contact_type'] : 'Contribution';
+    $customFields = CRM_Core_BAO_CustomField::getFields($customDataType);
+    // @todo call formatDateFields & move custom data handling there.
+    // Also note error handling for dates is currently in  _civicrm_api3_deprecated_formatted_param
+    // we should use the error handling in formatDateFields.
+    foreach ($params as $key => $val) {
+      // @todo - call formatDateFields instead.
+      if ($val) {
+        switch ($key) {
+          case 'receive_date':
+          case 'cancel_date':
+          case 'receipt_date':
+          case 'thankyou_date':
+            $params[$key] = CRM_Utils_Date::formatDate($params[$key], $dateType);
+            break;
+
+          case 'pledge_payment':
+            $params[$key] = CRM_Utils_String::strtobool($val);
+            break;
+
+        }
+        if ($customFieldID = CRM_Core_BAO_CustomField::getKeyID($key)) {
+          if ($customFields[$customFieldID]['data_type'] == 'Date') {
+            CRM_Contact_Import_Parser_Contact::formatCustomDate($params, $params, $dateType, $key);
+            unset($params[$key]);
```

Comment:
@eileenmcnaughton I'm tracking down an issue where contribution's custom fields typeof **date** is not being imported from a csv. The investigation led me to this line. I don't see (if there's a reason) why you are removing the value  to be imported of the custom field if it's date type.. am I missing anything here? tx!

Reply:
see **comment 2** above ^^